modules/avahi/avahi.c

```diff
@@ -0,0 +1,347 @@
+/**
+ * @file avahi.c Avahi Zeroconf Module
+ *
+ * Copyright (C) 2010 Creytiv.com
+ * Copyright (C) 2017 Jonathan Sieber
+ */
+
+/**
+ * @defgroup avahi avahi
+ *
+ * This module implements DNS Service Discovery via Avahi Client API
+ * It does 2 things:
+ * 1) Announce _sipuri._udp resource for the main UA (under the local IP)
+ * 2) Fills contact list with discovered hosts
+ *
+ * NOTE: This module is experimental.
+ *
+ */
+
+#include <re.h>
+#include <baresip.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <avahi-common/simple-watch.h>
+#include <avahi-client/lookup.h>
+#include <avahi-client/publish.h>
+#include <avahi-common/error.h>
+
+/* for if_nametoindex */
+#include <net/if.h>
+
+/* gethostname, getaddrinfo */
+#define __USE_XOPEN2K
+#include <unistd.h>
+#include <netdb.h>
+
+
+struct avahi_st {
+	AvahiSimplePoll* poll;
+	AvahiClient* client;
+	AvahiEntryGroup* group;
+	AvahiServiceBrowser* browser;
+	struct ua* local_ua;
```

Comment:
the destructor does not mem_deref the local_ua object. is there any good reason
for not doing that ?